app/models/message.rb

```diff
@@ -23,32 +23,54 @@ class Message < ActiveRecord::Base
 
   scope :with_eager_load, -> { includes([:sender]) }
   scope :non_private, -> { where(is_private: false) }
-  scope :donor_messages, ->(donor_id) { joins(:offer).where(offers: { created_by_id: donor_id }, is_private: false) }
+  scope :offer, -> { joins("INNER JOIN offers ON messages.messageable_id = offers.id and messages.messageable_type = 'Offer'") }
```

Comment:
we have a `belongs_to :offer` above, that can be adjusted to understand polymorphism rather than needing a scope.